src/globus_sdk/experimental/globus_app/authorizer_factory.py

```diff
@@ -110,6 +111,11 @@ class AccessTokenAuthorizerFactory(AuthorizerFactory[AccessTokenAuthorizer]):
     def __init__(self, token_storage: ValidatingTokenStorage) -> None:
         super().__init__(token_storage)
         self._cached_authorizer_expiration: dict[str, int] = {}
+        # inject a validator which ensures that tokens have an expiration time
+        # in the future
+        self.token_storage.after_retrieve_validators.insert(
+            0, self._check_expires_at_seconds
+        )
```

Comment:
The fact that authorizerfactory-specific attachments live in this module doesn't actually seem  necessary now that validators exist on their own.

I'd advocate that whatever context is building out the list of validators should own this too. Should make for simpler composition heuristics if validators only get added at one point.

Reply:
I think the changes I applied satisfy this pretty well. The validators exist independently and are associated in UserApp with the relevant authorizer factories.